### ISSUE TITLE
qtile freeze when XF86AUDIO key press

### ISSUE THREAD

**Issue by Jeff-WuYo** [qtile/qtile#4720]
### Issue description

I using qtile via xinitrc, I've configure the hotkey in config.py
```
Key([], "XF86AudioPlay", lazy.spawn("playerctl play-pause")),
Key([], "XF86AudioNext", lazy.spawn("playerctl next")),
Key([], "XF86AudioPrev", lazy.spawn("playerctl previous")),
Key([], "XF86AudioStop", lazy.spawn("playerctl stop")),
```
Any time I press any XF86Audioxxxx key, qtile freeze. I can still move my mouse, but couldn't control anything on this xorg session.  It's similar to #4505 .

[Here are xorg.0.log](https://pastebin.com/r3fgQELm), the log is too long for github comment.

And here are some journal output.
```
Mar 10 22:34:20 arch-z490-g org.a11y.Bus[9651]: X connection to :0 broken (explicit kill or server shutdown).
Mar 10 22:34:20 arch-z490-g systemd[1]: Started Process Core Dump (PID 26500/UID 0).
Mar 10 22:34:20 arch-z490-g systemd-coredump[26507]: [🡕] Process 9638 (picom) of user 1259 dumped core.

                                                     Stack trace of thread 9638:
                                                     #0  0x0000702d27a2f32c n/a (libc.so.6 + 0x8d32c)
                                                     #1  0x0000702d279de6c8 raise (libc.so.6 + 0x3c6c8)
                                                     #2  0x0000702d279c64b8 abort (libc.so.6 + 0x244b8)
                                                     #3  0x000062a63acd50b5 n/a (picom + 0xb0b5)
                                                     #4  0x000062a63ace9bc9 n/a (picom + 0x1fbc9)
                                                     #5  0x000062a63ad0cfaf n/a (picom + 0x42faf)
                                                     #6  0x000062a63acfdc2e n/a (picom + 0x33c2e)
                                                     #7  0x000062a63ad01ce4 n/a (picom + 0x37ce4)
                                                     #8  0x000062a63ad12906 n/a (picom + 0x48906)
                                                     #9  0x000062a63acd864c n/a (picom + 0xe64c)
                                                     #10 0x0000702d280730cb ev_invoke_pending (libev.so.4 + 0x50cb)
                                                     #11 0x0000702d28076d10 ev_run (libev.so.4 + 0x8d10)
                                                     #12 0x000062a63acd5582 n/a (picom + 0xb582)
                                                     #13 0x0000702d279c7cd0 n/a (libc.so.6 + 0x25cd0)
                                                     #14 0x0000702d279c7d8a __libc_start_main (libc.so.6 + 0x25d8a)
                                                     #15 0x000062a63acd6aa5 n/a (picom + 0xcaa5)
                                                     ELF object binary architecture: AMD x86-64
Mar 10 22:34:20 arch-z490-g systemd[1]: systemd-coredump@3-26500-0.service: Deactivated successfully.
```

### Version

0.24.0

### Backend

X11 (default)

### Config

```python
[My config file is on github public](https://github.com/Jeff-WuYo/Dot_file/blob/main/dot_config/qtile/config.py)
```

### Logs

```text
It only logged after I terminated xorg via other tty session. 

2024-03-10 22:34:21,297 ERROR libqtile loop.py:_handle_exception():L62 Exception in event loop:
Traceback (most recent call last):
  File "/usr/lib/python3.11/asyncio/events.py", line 84, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/lib/python3.11/site-packages/libqtile/core/manager.py", line 941, in f
    self.core.flush()
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/core.py", line 896, in flush
    self.conn.flush()
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/xcbq.py", line 613, in flush
    return self.conn.flush()
           ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 581, in wrapper
    self.invalid()
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 567, in invalid
    raise ConnectionException(err)
xcffib.ConnectionException: xcb connection errors because of socket, pipe and other stream errors.
2024-03-10 22:34:21,310 WARNING libqtile core.py:_xpoll():L366 Shutting down due to disconnection from X server
2024-03-10 22:34:21,311 WARNING libqtile core.py:graceful_shutdown():L908 Server disconnected, couldn't close windows gracefully.
2024-03-10 22:34:21,343 ERROR libqtile loop.py:_handle_exception():L62 Exception in event loop:
Traceback (most recent call last):
  File "/usr/lib/python3.11/asyncio/events.py", line 84, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/lib/python3.11/site-packages/libqtile/core/manager.py", line 959, in f
    self.core.flush()
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/core.py", line 896, in flush
    self.conn.flush()
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/xcbq.py", line 613, in flush
    return self.conn.flush()
           ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 577, in wrapper
    self.invalid()
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 567, in invalid
    raise ConnectionException(err)
xcffib.ConnectionException: xcb connection errors because of socket, pipe and other stream errors.
2024-03-10 22:34:21,356 ERROR libqtile base.py:_wrapper():L397 got exception from widget timer
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 395, in _wrapper
    method(*method_args)
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 786, in timer_setup
    update_interval = self.tick()
                      ^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 813, in tick
    self.update(text)
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 760, in update
    self.draw()
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 662, in draw
    self.drawer.draw(
  File "/usr/lib/python3.11/site-packages/libqtile/backend/base/drawer.py", line 216, in draw
    self._draw(
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/drawer.py", line 194, in _draw
    self.qtile.core.conn.conn.core.CopyArea(
  File "/usr/lib/python3.11/site-packages/xcffib/xproto.py", line 3152, in CopyArea
    return self.send_request(62, buf, is_checked=is_checked)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 389, in send_request
    seq = self.conn.send_request(flags, xcb_parts + 2, xcb_req)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 577, in wrapper
    self.invalid()
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 567, in invalid
    raise ConnectionException(err)
xcffib.ConnectionException: xcb connection errors because of socket, pipe and other stream errors.
2024-03-10 22:34:21,358 ERROR libqtile loop.py:_handle_exception():L62 Exception in event loop:
Traceback (most recent call last):
  File "/usr/lib/python3.11/asyncio/events.py", line 84, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/lib/python3.11/site-packages/libqtile/core/manager.py", line 959, in f
    self.core.flush()
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/core.py", line 896, in flush
    self.conn.flush()
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/xcbq.py", line 613, in flush
    return self.conn.flush()
           ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 577, in wrapper
    self.invalid()
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 567, in invalid
    raise ConnectionException(err)
xcffib.ConnectionException: xcb connection errors because of socket, pipe and other stream errors.
2024-03-10 22:34:21,359 ERROR libqtile loop.py:_handle_exception():L62 Exception in event loop:
Traceback (most recent call last):
  File "/usr/lib/python3.11/asyncio/events.py", line 84, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/lib/python3.11/site-packages/libqtile/core/manager.py", line 959, in f
    self.core.flush()
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/core.py", line 896, in flush
    self.conn.flush()
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/xcbq.py", line 613, in flush
    return self.conn.flush()
           ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 577, in wrapper
    self.invalid()
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 567, in invalid
    raise ConnectionException(err)
xcffib.ConnectionException: xcb connection errors because of socket, pipe and other stream errors.
2024-03-10 22:34:21,360 ERROR libqtile base.py:_wrapper():L397 got exception from widget timer
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 395, in _wrapper
    method(*method_args)
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 786, in timer_setup
    update_interval = self.tick()
                      ^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/libqtile/widget/clock.py", line 79, in tick
    self.update(self.poll())
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 760, in update
    self.draw()
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 662, in draw
    self.drawer.draw(
  File "/usr/lib/python3.11/site-packages/libqtile/backend/base/drawer.py", line 216, in draw
    self._draw(
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/drawer.py", line 194, in _draw
    self.qtile.core.conn.conn.core.CopyArea(
  File "/usr/lib/python3.11/site-packages/xcffib/xproto.py", line 3152, in CopyArea
    return self.send_request(62, buf, is_checked=is_checked)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 389, in send_request
    seq = self.conn.send_request(flags, xcb_parts + 2, xcb_req)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 577, in wrapper
    self.invalid()
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 567, in invalid
    raise ConnectionException(err)
xcffib.ConnectionException: xcb connection errors because of socket, pipe and other stream errors.
2024-03-10 22:34:21,361 ERROR libqtile loop.py:_handle_exception():L62 Exception in event loop:
Traceback (most recent call last):
  File "/usr/lib/python3.11/asyncio/events.py", line 84, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/lib/python3.11/site-packages/libqtile/core/manager.py", line 959, in f
    self.core.flush()
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/core.py", line 896, in flush
    self.conn.flush()
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/xcbq.py", line 613, in flush
    return self.conn.flush()
           ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 577, in wrapper
    self.invalid()
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 567, in invalid
    raise ConnectionException(err)
xcffib.ConnectionException: xcb connection errors because of socket, pipe and other stream errors.
2024-03-10 22:34:21,362 ERROR libqtile base.py:_wrapper():L397 got exception from widget timer
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 395, in _wrapper
    method(*method_args)
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 786, in timer_setup
    update_interval = self.tick()
                      ^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/libqtile/widget/clock.py", line 79, in tick
    self.update(self.poll())
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 760, in update
    self.draw()
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 662, in draw
    self.drawer.draw(
  File "/usr/lib/python3.11/site-packages/libqtile/backend/base/drawer.py", line 216, in draw
    self._draw(
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/drawer.py", line 194, in _draw
    self.qtile.core.conn.conn.core.CopyArea(
  File "/usr/lib/python3.11/site-packages/xcffib/xproto.py", line 3152, in CopyArea
    return self.send_request(62, buf, is_checked=is_checked)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 389, in send_request
    seq = self.conn.send_request(flags, xcb_parts + 2, xcb_req)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 577, in wrapper
    self.invalid()
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 567, in invalid
    raise ConnectionException(err)
xcffib.ConnectionException: xcb connection errors because of socket, pipe and other stream errors.
2024-03-10 22:34:21,362 ERROR libqtile loop.py:_handle_exception():L62 Exception in event loop:
Traceback (most recent call last):
  File "/usr/lib/python3.11/asyncio/events.py", line 84, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/lib/python3.11/site-packages/libqtile/core/manager.py", line 959, in f
    self.core.flush()
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/core.py", line 896, in flush
    self.conn.flush()
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/xcbq.py", line 613, in flush
    return self.conn.flush()
           ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 577, in wrapper
    self.invalid()
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 567, in invalid
    raise ConnectionException(err)
xcffib.ConnectionException: xcb connection errors because of socket, pipe and other stream errors.
2024-03-10 22:34:21,363 ERROR libqtile loop.py:_handle_exception():L62 Exception in event loop:
Traceback (most recent call last):
  File "/usr/lib/python3.11/asyncio/events.py", line 84, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/lib/python3.11/site-packages/libqtile/core/manager.py", line 959, in f
    self.core.flush()
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/core.py", line 896, in flush
    self.conn.flush()
  File "/usr/lib/python3.11/site-packages/libqtile/backend/x11/xcbq.py", line 613, in flush
    return self.conn.flush()
           ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 577, in wrapper
    self.invalid()
  File "/usr/lib/python3.11/site-packages/xcffib/__init__.py", line 567, in invalid
    raise ConnectionException(err)
xcffib.ConnectionException: xcb connection errors because of socket, pipe and other stream errors.
2024-03-10 22:34:21,366 ERROR libqtile base.py:on_done():L856 Failed to reschedule timer for cpu.
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 850, in on_done
    self.update(result)
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 754, in update
    old_width = self.layout.width
                ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/libqtile/backend/base/drawer.py", line 421, in width
    return self.layout.get_pixel_size()[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/libqtile/pangocffi.py", line 137, in get_pixel_size
    pango.pango_layout_get_pixel_size(self._pointer, width, height)
TypeError: initializer for ctype 'PangoLayout *' must be a cdata pointer, not NoneType
2024-03-10 22:34:21,366 ERROR libqtile base.py:on_done():L856 Failed to reschedule timer for memory.
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 850, in on_done
    self.update(result)
  File "/usr/lib/python3.11/site-packages/libqtile/widget/base.py", line 754, in update
    old_width = self.layout.width
                ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/libqtile/backend/base/drawer.py", line 421, in width
    return self.layout.get_pixel_size()[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/libqtile/pangocffi.py", line 137, in get_pixel_size
    pango.pango_layout_get_pixel_size(self._pointer, width, height)
TypeError: initializer for ctype 'PangoLayout *' must be a cdata pointer, not NoneType
2024-03-10 22:34:21,368 WARNING libqtile lifecycle.py:_atexit():L37 Qtile will now terminate
```
```


### Required

- [X] I have searched past issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by elParaguayo** [comment]
I use those keybindings and commands without issue so I'm a bit confused what's causing your issue.

The coredump suggests that picom is crashing so can you see if you still get the freeze if you don't run picom?

**Comment by Jeff-WuYo** [comment]
No, still freeze, although there's no picom error in the journal if I don't run picom. I still couldn't figure out what's the reason.

**Comment by tych0** [comment]
Seems like the X server is crashing? Do you see anything anywhere about that?

**Comment by Jeff-WuYo** [comment]
I found a [similar issue](https://github.com/ch11ng/exwm/issues/210#issuecomment-427869226). It seems there's nothing wrong with qtile, it's `dbus-launch` the culprit. I was testing how to launch fcitx5 properly, and there's a suggestion putting `eval "$(dbus-launch --sh-syntax --exit-with-session)"` at the start of .xinitrc, which is what I did. Removing the line resolve the issue. Sorry for the inconvenience, closing.
